packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -228,16 +244,40 @@ abstract class CachingAssetBundle extends AssetBundle {
     return completer.future;
   }
 
+  /// Retrieve bytedata from the asset bundle, parse it with the given function,
+  /// and return the function's result.
+  ///
+  /// The result of parsing the bytedata is cached (the bytedata itself is not). 
+  /// For any given `key`, the `decoder` is only run the first time.
+  ///
+  /// Once the value has been parsed, the future returned by this function for
+  /// subsequent calls will be a [SynchronousFuture], which resolves its
+  /// callback synchronously.
+  @override
+  Future<T> loadStructuredDataBinary<T>(String key, Future<T> Function(ByteData data) decoder) async {
```

Comment:
Because this method is `async` you will still need to wait an extra event loop even if it is cached. Write it more like loadStructuredData above